src/SFML/Window/WindowBase.cpp

```diff
@@ -207,18 +211,67 @@ void WindowBase::setSize(const Vector2u& size)
 {
     if (m_impl)
     {
-        m_impl->setSize(size);
+        // Constrain requested size within minimum and maximum bounds
+        const auto minimumSize = m_impl->getMinimumSize().value_or(sf::Vector2u());
+        const auto maximumSize = m_impl->getMaximumSize().value_or(
+            sf::Vector2u(std::numeric_limits<unsigned int>::max(), std::numeric_limits<unsigned int>::max()));
+        const auto width  = std::clamp(size.x, minimumSize.x, maximumSize.x);
+        const auto height = std::clamp(size.y, minimumSize.y, maximumSize.y);
+
+        // Do nothing if requested size matches current size
+        const sf::Vector2u clampedSize(width, height);
+        if (clampedSize == m_size)
+            return;
+
+        m_impl->setSize(clampedSize);
 
         // Cache the new size
-        m_size.x = size.x;
-        m_size.y = size.y;
+        m_size = clampedSize;
 
         // Notify the derived class
         onResize();
     }
 }
 
 
+////////////////////////////////////////////////////////////
+void WindowBase::setMinimumSize(const std::optional<Vector2u>& minimumSize)
+{
+    if (m_impl)
+    {
+        [[maybe_unused]] const auto minimumSizeIsLessThanMaximumSize = [this, minimumSize]()
+        {
+            if (!minimumSize.has_value() || !m_impl->getMaximumSize().has_value())
+                return true;
+            return minimumSize->x < m_impl->getMaximumSize()->x && minimumSize->y < m_impl->getMaximumSize()->y;
+        };
+        assert(minimumSizeIsLessThanMaximumSize() && "Minimum size must be less than maximum size");
+
+        m_impl->setMinimumSize(minimumSize);
+        setSize(getSize());
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void WindowBase::setMaximumSize(const std::optional<Vector2u>& maximumSize)
+{
+    if (m_impl)
+    {
+        [[maybe_unused]] const auto maximumSizeIsGreaterThanMinimumSize = [this, maximumSize]()
+        {
+            if (!maximumSize.has_value() || !m_impl->getMinimumSize().has_value())
+                return true;
+            return maximumSize->x > m_impl->getMinimumSize()->x && maximumSize->y > m_impl->getMinimumSize()->y;
+        };
+        assert(maximumSizeIsGreaterThanMinimumSize() && "Maximum size must be greater than minimum size");
```

Comment:
I'm not so sure we should assert here. I think ignoring and outputting a descript error would be better.

Output examples:
"Maximum size (200x400) is smaller than minimum size (300x200)! Ignoring"
"Minimum size (900x700) is larger than maximum size (800x600)! Ignoring"

Make sure this goes to stderr.

It would probably look something like this:
```c++
if (!maximumSizeIsGreaterThanMinimumSize())
{
    std::cerr << "Maximum size (" << maximumSize.x << "x" << maximumSize.y << 
        ") is smaller than minimum size (" << m_impl->getMinimumSize()->x << 
        "x" << m_impl->getMinimumSize()->y << ")! Ignoring\n";
    return;
}
```

Reply:
Also, inverting the output and renaming the `maximumSizeIsGreaterThanMinimumSize` function to `maximumSizeIsSmallerthanMaximumSize` may be more optimal in this situation.

Reply:
https://github.com/SFML/SFML/pull/2519#discussion_r1181098760

Comment on this one. I'm resolving this conversation to avoid people splitting off conversations between the two reviews.